components/channel_layout/app_route/app_route.jsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Pluggable from 'plugins/pluggable';
+import ChannelHeader from 'components/channel_header';
+
+export default class AppRouter extends React.PureComponent {
+    static propTypes = {
+
+        /*
```

Comment:
Perfectly fine to leave out of scope, but I wonder if adopting typescript would help with the shapes here.

Reply:
don't know how to answer that